src/Service/Program.cs

```diff
@@ -27,6 +27,13 @@ public class Program
 
         public static void Main(string[] args)
         {
+            // Compare current version of DAB with latest (non-rc) version in NuGet. 
+            VersionChecker.GetVersions(out string? latestVersion, out string? currentVersion);
+            if (!string.IsNullOrEmpty(latestVersion) && latestVersion != currentVersion)
+            {
+                Console.Error.WriteLine($"A newer version of Data API builder is available. {currentVersion} -> {latestVersion}");
```

Comment:
Should we log a warning instead of an error? Would this affect the docker scenario too?

Reply:
Since moving it to `dab validate` I have the logger and updated it to Warn.